src/pages/view-test-result/view-test-result.ts

```diff
@@ -333,6 +335,58 @@ export class ViewTestResultPage extends BasePageComponent implements OnInit {
     return getEyesightTestSeriousFaultAndComment(eyesightTest).map(this.parseResult);
   }
 
+  getRekeyDetails(): RekeyDetailsModel {
+    if (!this.testResult) {
+      return null;
+    }
+
+    const testDate: DateTime = new DateTime(this.testResult.journalData.testSlotAttributes.start);
+    const rekeyDate: DateTime = new DateTime(this.testResult.rekeyDate);
+
+    return {
+      scheduledStaffNumber: this.testResult.examinerBooked,
+      conductedStaffNumber: this.testResult.examinerConducted,
+      testDate: testDate.format('dddd Do MMMM YYYY'),
+      rekeyedStaffNumber: this.testResult.examinerKeyed,
+      rekeyDate: rekeyDate.format('dddd Do MMMM YYYY'),
+    };
+  }
+
+  getRekeyReason(): RekeyReasonModel {
+    if (!this.testResult) {
+      return null;
+    }
+
+    const getIpadIssueDisplayText = (reasonType): string => {
+
+      const ipadIssueText = {
+        broken: 'Broken',
+        lost: 'Lost',
+        technicalFault: 'Technical fault',
+        stolen: 'Stolen',
+      };
+
+      let value = '';
+
+      for (const prop in reasonType) {
```

Comment:
Don't think that we need a loop here as we just want to return a string considering the properties on reasonType and we know that there are only 4 of them. Usage of if statements would be simple and clean.

Reply:
Ok, happy to change to if's if you think that's cleaner.